packages/block-library/src/missing/index.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { RawHTML, Fragment } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { getBlockType, createBlock } from '@wordpress/blocks';
+import { withDispatch } from '@wordpress/data';
+import { Warning } from '@wordpress/editor';
+
+function MissingBlockWarning( { attributes, convertToHTML } ) {
+	const { originalName, originalUndelimitedContent } = attributes;
+	const hasContent = !! originalUndelimitedContent;
+	const hasHTMLBlock = getBlockType( 'core/html' );
+
+	const actions = [];
+	let messageHTML;
+	if ( hasContent && hasHTMLBlock ) {
+		messageHTML = sprintf(
+			__( 'Your site doesn\'t include support for the <code>%s</code> block. You can leave this block intact, convert its content to a Custom HTML block, or remove it entirely.' ),
+			originalName
+		);
+		actions.push(
+			<Button key="convert" onClick={ convertToHTML } isLarge isPrimary>
+				{ __( 'Keep as HTML' ) }
+			</Button>
+		);
+	} else {
+		messageHTML = sprintf(
+			__( 'Your site doesn\'t include support for the <code>%s</code> block. You can leave this block intact or remove it entirely.' ),
+			originalName
+		);
+	}
+
+	return (
+		<Fragment>
+			<Warning actions={ actions }>
+				<span dangerouslySetInnerHTML={ { __html: messageHTML } } />
```

Comment:
In this case, dangerous is _actually dangerous_. Localized strings should be considered equivalent to user input for sanitization purposes. 

> since translations should not be considered trusted strings, be sure to sanitize the result before echoing.

https://codex.wordpress.org/I18n_for_WordPress_Developers#HTML

Related: #9846

Reply:
-> #10626

Reply:
Thanks very much for catching this, @aduth, and for alerting me to the need for care here.

@mcsf, thank you for the fix. <3

Reply:
>FWIW this isn't entirely accurate. WordPress.org treats stranslations as trusted strings.

Is the Codex wrong then? Or is the article not applicable for core development?

Reply:
Both I guess :-)